src/V3Partition.cpp

```diff
@@ -2614,6 +2614,39 @@ void V3Partition::hashGraphDebug(const V3Graph* graphp, const char* debugName) {
 uint32_t V3Partition::setupMTaskDeps(V3Graph* mtasksp) {
     uint32_t totalGraphCost = 0;
 
+    // The fine-grained dependency graph of MTaskMoveVertex vertices is a bipartite graph of:
+    // - 1. MTaskMoveVertex instances containing logic via OrderLogicVertex
+    //      (MTaskMoveVertex::logicp() != nullptr)
+    // - 2. MTaskMoveVertex instances containing an (OrderVarVertex, domain) pair
+    // Our goal is to order the logic vertices. The second type of variable/domain vertices only
+    // carry dependencies and are eventually discarded. In order to reduce the working set size of
+    // PartContraction, we 'bypass' and not create LogicMTask vertices for the variable vertices,
+    // and instead add the transitive dependencies directly, but only if adding the transitive
+    // edges directly does not require more dependency edges than keeping the intermediate vertex.
+    // That is, we bypass a variable vertex if fanIn * fanOut <= fanIn + fanOut. This can only
+    // be true if fanIn or fanOut are 1, or if they are both 2. This can cause significant
+    // reduction in working set size.
+    const auto bypass = [](MTaskMoveVertex* mvtxp) -> bool {
```

Comment:
I know our styles differ here, but I think this is clearer as a separate "bypassOk" function.  Then this and the surrounding function both fit in one screen which helps understanding.